services/core/shared/core/compat/sdk_v4/transactions.js

```diff
@@ -61,6 +62,19 @@ const getTransactions = async params => {
 		if (response.meta) transactions.meta = response.meta;
 	}
 
+	transactions.data = await BluebirdPromise.map(
+		transactions.data,
+		async transaction => {
+			if (!transaction.timestamp) {
+				const txBlock = (await coreApi.getBlocks({ height: transaction.height })).data[0];
+				transaction.timestamp = txBlock.timestamp;
+			}
```

Comment:
Is there any case in SDKv4 that the transaction time is there already?

Reply:
In case of (params.fromTimestamp ||  params.totimeStamp), we already resolve timestamp when we call function `getTransactionsByBlock`. So, just to avoid an unnecessary call i added this check.

```
	if (params.fromTimestamp || params.toTimestamp) {
		const blocks = await coreApi.getBlocks(params);
		transactions.meta.count = 0;
		await BluebirdPromise.map(
			blocks.data,
			async (block) => {
				const blkTransactions = await getTransactionsByBlock(block);
				transactions.data = transactions.data.concat(blkTransactions.data);
				transactions.meta.count += blkTransactions.meta.count;
			},
			{ concurrency: blocks.data.length },
		);
	}
```